### PR TITLE
Splitting input reference file

### PR DESCRIPTION
**Pull request by gq1**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/variantcalling/tree/main/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by gq1** [comment]
The full test on the farm took less than 2 hours to finish after splitting:
https://tower.nf/orgs/sanger-tol/workspaces/github-ci/watch/1idBZO6ooa0B5D
This time completely depend on the longest reference sequence, which is 1/3 of the whole fasta files.

19 sequences in total in the fasta file, split them into 8 files by putting the small sequence together.

File: subworkflows/local/deepvariant_caller.nf [hunk 1]
```diff
@@ -1,41 +1,66 @@
...
+    // fasta
+    reads_fasta.map{ meta, cram, crai, interval, fasta_file_name, fasta, fai -> [ fasta ] }
+            .set{ fasta }
 
+    // fai
+    reads_fasta.map{ meta, cram, crai, interval, fasta_file_name, fasta, fai -> [ fai ] }
+            .set{ fai }
+
+    // split fasta in compressed format, no gzi index file needed
+    gzi = []
```
Comment (priyanka-surana) [inline subworkflows/local/deepvariant_caller.nf:34]: What happens if someone provides the `gzi` file?

File: subworkflows/local/input_filter_split.nf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     .set{ split_fasta } 
+
+    // index split fasta files
+    SAMTOOLS_FAIDX ( split_fasta  )
+    ch_versions = ch_versions.mix( SAMTOOLS_FAIDX.out.versions.first() )
+    
+    // join fasta with corresponding fai file
+    split_fasta
+     .map{ meta, fasta -> [ fasta.baseName, fasta ] }
+     .join(
```
Comment (priyanka-surana) [inline subworkflows/local/input_filter_split.nf:61]: Why do the join only to split it in the next subworkflow?

File: subworkflows/local/input_filter_split.nf [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    // filter reads
+    SAMTOOLS_VIEW ( reads, fasta, [] )
+    ch_versions = ch_versions.mix ( SAMTOOLS_VIEW.out.versions.first() )
+    
+    // combine reads with splitted references
+    cram_crai_fasta_fai = SAMTOOLS_VIEW.out.cram
+                        .join(SAMTOOLS_VIEW.out.crai)
+                        .map { filtered_reads -> filtered_reads + [interval ?: []] }
+                        .combine( fasta_fai )
```
Comment (priyanka-surana) [inline subworkflows/local/input_filter_split.nf:75]: Same for this combine.

File: nextflow.config
Comment (priyanka-surana) [inline nextflow.config:1]: Only the `manifest` needs updating – add all contributors and put a bit more in description. Fix the triple quotes too.

**Review by priyanka-surana — CHANGES_REQUESTED** [review]
Works for me and looks good, but I have some questions/suggestions about the structure.

File: subworkflows/local/deepvariant_caller.nf (hunk 1 above)
Comment (gq1) [reply subworkflows/local/deepvariant_caller.nf:34]: This gzi file for the split fasta files, not for the original input fasta files.

input gzi file still needed if the given fasta file is compressed, that will be used by samtools view filtering.

**Review by priyanka-surana — APPROVED** [review]

**Pull request merged**